### ISSUE TITLE
when using 'children' but not 'parent' property in json model for jstree pushing to tree model doesn't update the jstree view

### ISSUE THREAD

**Issue by fastovezz** [ezraroi/ngJsTree#47]
Hi!

When I use json data for jsTree with children but without parent, like this:
{
  id          : "string" // will be autogenerated if omitted
  text        : "string" // node text
  icon        : "string" // string for custom
  state       : {
    opened    : boolean  // is the node open
    disabled  : boolean  // is the node disabled
    selected  : boolean  // is the node selected
  },
  children    : []  // array of strings or objects
  li_attr     : {}  // attributes for the generated LI node
  a_attr      : {}  // attributes for the generated A node
} // note there is no parent property in this json

-- then pushing to ngjJsTree's model doesn't rerender jsTree view/ incrementing version will do the trick, but I don't want to rerender whole tree if only one node is added.

I noticed that you use following code to add newly pushed note to the ngjsTree's model:
...
var parent = scope.tree.jstree(true).get_node(node.parent);
var res = scope.tree.jstree(true).create_node(parent, node, 'inside',function() {
...
-- so if node doesn't have parent property(as in my case) node wont' be added/

Using children and parent together is impossible as jstree prefers parent and ignores children as far as I understand.

**Comment by ezraroi** [comment]
will check this week

**Comment by fastovezz** [comment]
Thanks!
And one more thing. Your method:
```javascript
            var getTokens = function () {
                var result = [], token, el;
                if (arraySource) {
                    var array = angular.isFunction(arraySource) ? arraySource() : arraySource;
                    for (var i = 0, n = array.length; i < n; i++) {
                        el = array[i];
                        token = tokenFn(el);
                        map[token] = el;
                        result.push(token);
                    }
                }
                return result;
            };
```
which is used to watch model changes watches only first-level elements (doesn't take 'children' property into account).

**Comment by ezraroi** [comment]
Yes, because in the other syntax (the parent one), all are first level.
From what i saw, there is not `create_node` function in the jsTree that doesn't require parent node. Looks like to support this data format i will need to change many things, not sure i will have time soon. Sorry, you are welcome to fork and pr :)

**Comment by fastovezz** [comment]
ok, got it, thanks for your quick reply :)

**Comment by LorandLorincz** [comment]
+1. would need this.
